tests/e2e/apps/hop/app.go

```diff
@@ -34,13 +34,39 @@ import (
 	"google.golang.org/grpc"
 
 	"istio.io/istio/tests/e2e/apps/hop/config"
+	"istio.io/istio/tests/e2e/framework"
+	"istio.io/istio/tests/e2e/util"
 )
 
 var (
-	timeout = flag.Duration("timeout", 15*time.Second, "Request timeout")
-	version = flag.String("version", "", "Server version")
+	timeout     = flag.Duration("timeout", 15*time.Second, "Request timeout")
+	version     = flag.String("version", "", "Server version")
+	hopYamlTmpl = "tests/e2e/framework/testdata/hop.yam.tmpl"
 )
 
+// hopTemplate gathers template variable for hopYamlTmpl.
+type hopTemplate struct {
+	Deployment string
+	Service    string
+	HTTPPort   int
+	GRPCPort   int
+	Version    string
+}
+
+// NewHop instantiate a framework.App to be used by framework.AppManager.
+func NewHop(d, s, v string, h, g int) *framework.App {
+	return &framework.App{
+		AppYamlTemplate: util.GetResourcePath(hopYamlTmpl),
+		Template: &hopTemplate{
+			Deployment: d,
+			Service:    s,
```

Comment:
Please note that in the original Service was being set to d; seems like the original was a mistake, but am raising this so that we can be ware of any discrepancies.
<img width="567" alt="screen shot 2017-06-15 at 10 21 52 pm" src="https://user-images.githubusercontent.com/4898263/27211816-33511418-5219-11e7-829c-2c4e61ea56e8.png">